olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h

```diff
@@ -49,7 +50,18 @@ class CORE_API HttpResponse {
    */
   HttpResponse(int status, std::stringstream&& response)
       : status(status), response(std::move(response)) {}
-
+  /**
+   * @brief Creates the `HttpResponse` instance.
+   *
+   * @param status The HTTP status.
+   * @param response The response body.
+   * @param headers The headers of the responce.
+   */
+  HttpResponse(int status, std::stringstream&& response,
+               http::HeadersType&& headers)
```

Comment:
Maybe we can start moving with this also towards named parameter idiom same as for NetworkRequest.
So that we do not have to put everything in constructor anymore (which btw is an API break!).

Reply:
HttpResponse currently has all parameters public, without any setters. In this case I need to change this class completely to use named parameter idiom.

Reply:
Ok, let's leave it for now and do it later.